interchain-token-service/src/msg.rs

```diff
@@ -0,0 +1,45 @@
+use std::collections::HashMap;
+
+use axelarnet_gateway::AxelarExecutableMsg;
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use msgs_derive::EnsurePermissions;
+use router_api::{Address, ChainName, ChainNameRaw};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub governance_address: String,
+    pub admin_address: String,
+    pub chain_name: ChainNameRaw,
+    pub gateway_address: String,
+    pub its_addresses: HashMap<ChainName, Address>,
+}
+
+#[cw_serde]
+#[derive(EnsurePermissions)]
+pub enum ExecuteMsg {
+    #[permission(Specific(gateway))]
+    Execute(AxelarExecutableMsg),
+    #[permission(Governance)]
+    SetItsAddress { chain: ChainName, address: Address },
+    #[permission(Elevated)]
+    RemoveItsAddress { chain: ChainName },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    #[returns(ItsAddressResponse)]
+    SetItsAddress { chain: ChainName },
+    #[returns(AllItsAddressesResponse)]
+    AllItsAddresses {},
+}
+
+#[cw_serde]
+pub struct ItsAddressResponse {
+    pub address: Option<Address>,
+}
```

Comment:
Do we need this struct?

Reply:
Yeah, no point of wrapping. I'm used to axelar-core wrapper protobufs for query responses lol